From 5f89c8f49f6f44b2f4da8330748666f910b8f823 Mon Sep 17 00:00:00 2001
From: Masahiro Yamada <masahiroy@kernel.org>
Date: Sat, 22 Jul 2023 13:47:57 +0900
Subject: [PATCH 15/62] kbuild: rpm-pkg: derive the Version from
 %{KERNELRELEASE}

Avoid hard-coding the Version field in the generated spec file.

Signed-off-by: Masahiro Yamada <masahiroy@kernel.org>
---
 scripts/package/mkspec | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/scripts/package/mkspec b/scripts/package/mkspec
index 22e290d23..783e1997d 100755
--- a/scripts/package/mkspec
+++ b/scripts/package/mkspec
@@ -27,8 +27,6 @@ else
 	M=DEL
 fi
 
-__KERNELRELEASE=$(echo $KERNELRELEASE | sed -e "s/-/_/g")
-
 # We can label the here-doc lines for conditional output to the spec file
 #
 # Labels:
@@ -43,7 +41,7 @@ sed -e '/^DEL/d' -e 's/^\t*//' <<EOF
 
 	Name: kernel
 	Summary: The Linux Kernel
-	Version: $__KERNELRELEASE
+	Version: %(echo %{KERNELRELEASE} | sed -e 's/-/_/g')
 	Release: $(cat .version 2>/dev/null || echo 1)
 	License: GPL
 	Group: System Environment/Kernel
-- 
2.42.0

